serve.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/weaveworks/footloose/pkg/api"
+)
+
+var serveCmd = &cobra.Command{
+	Use:   "serve",
+	Short: "Launch a footloose server",
+	RunE:  serve,
+}
+
+var serveOptions struct {
+	listen string
+}
+
+func baseURI(addr string) (string, error) {
+	host, port, err := net.SplitHostPort(addr)
+	if err != nil {
+		return "", err
+	}
+	if host == "" || host == "0.0.0.0" || host == "[::]" {
+		host = "localhost"
+	}
+	return fmt.Sprintf("http://%s:%s", host, port), nil
+}
+
+func init() {
+	serveCmd.Flags().StringVarP(&serveOptions.listen, "listen", "l", ":2444", "Cluster configuration file")
+	footloose.AddCommand(serveCmd)
+}
+
+func serve(cmd *cobra.Command, args []string) error {
+	opts := &serveOptions
+
+	baseURI, err := baseURI(opts.listen)
+	if err != nil {
+		return errors.Wrapf(err, "invalid listen address '%s'", opts.listen)
+	}
+
+	api := api.New(baseURI)
```

Comment:
IIUC, `api` holds its own in-memory data store with clusters and machines that's initialized empty on command start.
One problem here is that machines survive past `footloose serve` execution (and are still completely valid Footloose machines), but they're inaccessible/leaked (from the API perspective).

Maybe the API should either maintain state between `footloose serve` executions, or clean up the machines on `footloose server` termination?

Reply:
Because the state is already in the docker daemon, I was thinking `footloose server` could read back the initial state from it at start up (but thought it would be something for a later PR).

Reply:
(But yes, I 100% agree something like this is needed)